File: split/printhex.pas
procedure printhex(n: integer);
var
  k: 0..22;
begin
  k := 0;
  printchar(34);
  repeat
    dig[k] := n mod 16;
    n := n div 16;
    k := k + 1;
  until n = 0;
  printthedigs(k);
end;

File: cpp/src/printhex.cpp
#include "printhex.h"
#include "printchar.h"
#include "printthedigs.h"

void printhex(int n)
{
	int k = 0;
	printchar('"');
	do
	{
		dig[k] = n%16;
		n /= 16;
		k++;
	}
	while (n);
	printthedigs(k);
}
